torch/distributed/_tensor/op_schema.py

```diff
@@ -65,7 +69,69 @@ class PlacementStrategy:
     # this operator it might have multiple placement strategies
     redistribute_cost: Optional[List[List[float]]] = None
 
-    def pretty_print_placements(self, placements):
+    def mesh_shape(self) -> Tuple[int, ...]:
+        if isinstance(self.output_spec, DTensorSpec):
+            return self.output_spec.mesh.shape
+        else:
+            if isinstance(self.output_spec[0], DTensorSpec):
+                return self.output_spec[0].mesh.shape
+            else:
+                return tuple()
+
+    def output_ndim(self) -> int:
+        if isinstance(self.output_spec, DTensorSpec):
+            return self.output_spec.ndim
+        else:
+            if isinstance(self.output_spec[0], DTensorSpec):
+                return self.output_spec[0].ndim
+            else:
+                return 0
+
+    def output_shape(self) -> torch.Size:
+        if isinstance(self.output_spec, DTensorSpec):
+            return self.output_spec.shape
+        else:
+            if isinstance(self.output_spec[0], DTensorSpec):
+                return self.output_spec[0].shape
+            else:
+                return torch.Size([])
+
+    def num_shards(self) -> int:
+        if isinstance(self.output_spec, DTensorSpec):
+            return self.output_spec.num_shards
+        else:
+            if isinstance(self.output_spec[0], DTensorSpec):
+                return self.output_spec[0].num_shards
+            else:
+                return 0
+
+    def pretty_print_output_spec(self) -> str:
+        if isinstance(self.output_spec, DTensorSpec):
+            return str(self.output_spec)
+        else:
+            output_spec_str = []
+            for spec in self.output_spec:
+                if spec is None:
+                    output_spec_str.append("None")
+                else:
+                    output_spec_str.append(str(spec))
+            return "(" + ", ".join(output_spec_str) + ")"
+
+    def pretty_print_output_spec_placements(self) -> str:
```

Comment:
why we have two functions?

* pretty_print_output_spec
* pretty_print_output_spec_placements?

What's the difference here?

I think we should only expose 1 function like before: `pretty_print_placements`

Reply:
I want to keep `pretty_print_output_spec` because in `OpSchema.__str__` we need to get the `output_spec` string. Using `pretty_print_placements` to do that is quite verbose (10 extra lines in that `elif` branch).